lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp

```diff
@@ -643,17 +643,17 @@ LLDBMemoryReader::resolveRemoteAddress(uint64_t address) const {
   auto sec_file_address = sec->GetFileAddress();
   auto sec_load_address = sec->GetLoadBaseAddress(&m_process.GetTarget());
 
-  bool overflow = false;
-  auto slide =
-      llvm::SaturatingAdd(sec_load_address, -sec_file_address, &overflow);
-  if (overflow) {
+  if (sec_load_address < sec_file_address) {
     LLDB_LOG(log,
-             "[MemoryReader] section load address {0:x} - file address {1:x} "
-             "overflows",
+             "[MemoryReader] section load address {0:x} is smaller than "
+             "section file address {1:x}",
              sec_load_address, sec_file_address);
     return {};
   }
 
+  auto slide = sec_load_address - sec_file_address;
```

Comment:
FWIW I think this file is using too much `auto`. None of these types are obvious. For example, I think the overflow check would have stood out more if `sec_file_address` and `sec_load_address` above would be marked as `lldb::addr_t` (aka `uint64_t`).

Reply:
I agree, I replaced those auto with the types.